### ISSUE TITLE
"www" in links is hidden visually when it should not be

### ISSUE THREAD

**Issue by trwnh** [mastodon/mastodon#21190]
### Steps to reproduce the problem

Post a link for a URL on a `www` subdomain

### Expected behaviour

`www.` is shown

### Actual behaviour

`www.` is hidden

### Detailed description

Hiding the `https://` can make sense to reduce visual clutter, but hiding the `www.` changes the semantic meaning of the link. It is not a given that `www` and `@` (no subdomain) are equivalent or redirect to each other.

### Specifications

mastodon.social 4.0.2

**Comment by afontenot** [comment]
I agree that this is less than ideal but it does mirror the behavior of e.g. Chromium. I think the usual expectation for websites these days is that either www will redirect to @ or vice versa.

**Comment by ineffyble** [comment]
> It is not a given that www and @ (no subdomain) are equivalent or redirect to each other.

It's not technically enforced and there's no official standard, but I would say it's a widely followed (and assumed) convention in modern times.
